dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -3,102 +3,487 @@
-  const onChangeItemsPerPage = useCallback(
-    (pageSize) =>
-      setPagination((pagination) => ({
-        ...pagination,
-        pageSize,
-        pageIndex: 0,
-      })),
-    [setPagination]
-  );
-  const onChangePage = useCallback(
-    (pageIndex) => setPagination((pagination) => ({ ...pagination, pageIndex })),
-    [setPagination]
-  );
+  // ag-grid-react bindings
+  const gridRef = useRef<any | undefined>();
+  const gridRefFullScreen = useRef<any | undefined>();
+  const [pageSize, setPageSize] = useState(10);
+  const [columnVisibility, setColumnVisibility] = useState([]);
+  const [isFullScreen, setIsFullScreen] = useState<boolean>(false);
+  const [selectedRowDensity, setSelectedRowDensity] = useState({
+    icon: 'tableDensityNormal',
+    height: 40,
+    selected: true,
+  });
+  const defaultColDef = useMemo(() => {
+    return {
+      editable: true,
+      sortable: true,
+      resizable: true,
+      filter: true,
+      flex: 1,
+      minWidth: 100,
+      suppressMenu: true,
+    };
+  }, []);
 
-  // Sorting
-  const [sortingColumns, setSortingColumns] = useState([]);
-  const onSort = useCallback(
-    (sortingColumns) => {
-      setSortingColumns(sortingColumns);
+  const onPageSizeChanged = useCallback(
+    (val) => {
+      setPageSize(val);
+      gridRef.current.api.paginationSetPageSize(val);
+      setActivePage(0);
+      gridRef.current.api.paginationGoToPage(0);
+      if (isFullScreen) {
+        gridRefFullScreen.current.api.paginationSetPageSize(val);
+        gridRefFullScreen.current.api.paginationGoToPage(0);
+      }
     },
-    [setSortingColumns]
+    [isFullScreen]
   );
 
-  // Sort data
-  let data = useMemo(() => {
-    return [...raw_data].sort((a, b) => {
-      for (let i = 0; i < sortingColumns.length; i++) {
-        const column = sortingColumns[i];
-        const aValue = a[column.id];
-        const bValue = b[column.id];
+  const paginationNumberFormatter = useCallback((params) => {
+    return '[' + params.value.toLocaleString() + ']';
+  }, []);
 
-        if (aValue < bValue) return column.direction === 'asc' ? -1 : 1;
-        if (aValue > bValue) return column.direction === 'asc' ? 1 : -1;
+  const selectDensityHandler = useCallback((value) => {
+    setSelectedRowDensity({ ...value });
+    gridRef.current.api.forEachNode((rowNode) => {
+      if (rowNode.data) {
+        rowNode.setRowHeight(value.height);
       }
-
-      return 0;
     });
-  }, [sortingColumns]);
+    gridRef.current.api.onRowHeightChanged();
+  }, []);
 
-  // Pagination
-  data = useMemo(() => {
-    const rowStart = pagination.pageIndex * pagination.pageSize;
-    const rowEnd = Math.min(rowStart + pagination.pageSize, data.length);
-    return data.slice(rowStart, rowEnd);
-  }, [data, pagination]);
+  const columnVisiblityHandler = useCallback((visible: boolean, feild: any) => {
+    const isExisting = columnVisibility.findIndex((i) => i === feild);
```

Comment:
instead of `i` use some valid name like `column` or similar.

Reply:
fixed